v2/pkg/subscraping/utils.go

```diff
@@ -32,6 +33,7 @@ func PickRandom[T any](v []T, sourceName string) T {
 		gologger.Debug().Msgf("Cannot use the '%s' source because there was no API key/secret defined for it.", sourceName)
 		return result
 	}
+	rand.Seed(time.Now().UnixNano())
```

Comment:
this does not need to be in this function. you can just have an init() function and call it there once

Reply:
calling it each time is redundant and unncessary